src/Sylius/Component/Channel/Context/EnvBased/ChannelContext.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Channel\Context\EnvBased;
+
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Channel\Context\ChannelNotFoundException;
+use Sylius\Component\Channel\Model\ChannelInterface;
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+
+final class ChannelContext implements ChannelContextInterface
```

Comment:
Is it really necessary to have `EnvBased` in namespace instead of class name, like the `SingleChannelContext`?

Reply:
Any ideas? I thought if `EnvBased` is an accurate name 🤔. We could pass anything in the constructor, and the channel is set 🤔.

Reply:
`Sylius\Component\Channel\Context\EnvBasedChannelContext` should be enough 🤷‍♂️

Reply:
It's better for DX IMO.